File: ES01-Jakarta/src/main/java/it/distributedsystems/web/HelloServlet.java
package it.distributedsystems.web;

import it.distributedsystems.model.ejb.HelloBean;
import jakarta.servlet.ServletException;
import jakarta.servlet.annotation.WebServlet;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

@WebServlet(name = "HelloServlet", urlPatterns = {"hello"}, loadOnStartup = 1)
public class HelloServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.getWriter().print("Hello, World!");
    }

    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        HelloBean ejb = new HelloBean();
        String name = request.getParameter("name");
        if (name == null) name = "World";
        PrintWriter out = response.getWriter();
        out.println("Hello "+name+"!!!");
        out.println(ejb.toString());
    }
}

File: Ex1/src/main/kotlin/it/distributedsystems/model/web/HelloServlet.kt
package it.distributedsystems.model.web

import it.distributedsystems.model.ejb.HelloBean
import java.io.IOException
import javax.servlet.ServletException
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import kotlin.jvm.Throws

@WebServlet(name = "HelloServlet", urlPatterns = ["hello"], loadOnStartup = 1)
class HelloServlet : HttpServlet() {

    @Throws(ServletException::class, IOException::class)
    override fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        resp.writer.print("Hello, World!");
    }

    @Throws(ServletException::class, IOException::class)
    override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {
        val ejb = HelloBean()
        val name = req.getParameter("name") ?: "World"
        val out = resp.writer
        out.println("Hello ${name}!!!")
        out.println(ejb.toString())
    }

}